### PR TITLE
BASE: Code formatting cleanup pass of base

### PR DESCRIPTION
**Pull request by 8BitPimp**
This is a non-functional change, that cleans up some of the source formatting in base.
Please let me know if you spot anything that is not in keeping with established ScummVM format.

Also if anyone knows how to get github to display 4 space tabs, please let me know :)

### CODE REVIEW

File: base/commandLine.cpp [hunk 1]
```diff
@@ -54,8 +54,7 @@ static const char USAGE_STRING[] =
 	"%s: %s\n"
 	"Usage: %s [OPTIONS]... [GAME]\n"
 	"\n"
-	"Try '%s --help' for more options.\n"
-;
```
Comment (sev-) [inline base/commandLine.cpp]: This was done on purpose.

File: base/commandLine.cpp [hunk 2]
```diff
@@ -167,8 +166,7 @@ static const char HELP_STRING[] =
 #endif
 	"\n"
 	"The meaning of boolean long options can be inverted by prefixing them with\n"
-	"\"no-\", e.g. \"--no-aspect-ratio\".\n"
-;
+	"\"no-\", e.g. \"--no-aspect-ratio\".\n";
```
Comment (sev-) [inline base/commandLine.cpp]: Same here. The purpose is to minimize diffs on changes.

File: base/main.cpp [hunk 3]
```diff
@@ -153,24 +153,24 @@ static Common::Error runGame(const EnginePlugin *plugin, OSystem &system, const
 	// needed because otherwise the g_system->getSupportedFormats might return
 	// bad values.
 	g_system->beginGFXTransaction();
-		g_system->setGraphicsMode(ConfMan.get("gfx_mode").c_str());
```
Comment (sev-) [inline base/main.cpp]: This is done on purpose, to visually set up code which is within the transaction.

File: base/main.cpp [hunk 4]
```diff
@@ -282,19 +282,18 @@ static Common::Error runGame(const EnginePlugin *plugin, OSystem &system, const
 }
 
 static void setupGraphics(OSystem &system) {
-
 	system.beginGFXTransaction();
-		// Set the user specified graphics mode (if any).
-		system.setGraphicsMode(ConfMan.get("gfx_mode").c_str());
+	// Set the user specified graphics mode (if any).
```
Comment (sev-) [inline base/main.cpp]: Same here, these are inside a code block.

File: base/main.cpp [hunk 5]
```diff
@@ -551,17 +553,17 @@ extern "C" int scummvm_main(int argc, const char * const argv[]) {
 			#ifndef FORCE_RTL
 			if (result.getCode() == Common::kNoError && !g_system->getEventManager()->shouldRTL())
 				break;
-			#endif
+#endif
```
Comment (sev-) [inline base/main.cpp]: Now, the #endif does not match the indentation of the #ifndef. Thus, either have both unindented, or do not make this change at all.

File: base/main.cpp [hunk 6]
```diff
@@ -589,7 +591,8 @@ extern "C" int scummvm_main(int argc, const char * const argv[]) {
 				ConfMan.setActiveDomain("");
 			}
 
-			PluginManager::instance().loadAllPlugins(); // only for cached manager
+			// only for cached manager
```
Comment (sev-) [inline base/main.cpp]: Why this change?

File: base/plugins.cpp [hunk 7]
```diff
@@ -57,89 +52,85 @@ class StaticPlugin : public Plugin {
...
+	~StaticPluginProvider() {}
 
 	virtual PluginList getPlugins() {
 		PluginList pl;
 
-		#define LINK_PLUGIN(ID) \
-			extern PluginType g_##ID##_type; \
-			extern PluginObject *g_##ID##_getObject(); \
-			pl.push_back(new StaticPlugin(g_##ID##_getObject(), g_##ID##_type));
+#define LINK_PLUGIN(ID) \
```
Comment (sev-) [inline base/plugins.cpp]: And this badly breaks visuals of the code.

File: base/plugins.h [hunk 8]
```diff
@@ -58,38 +56,34 @@
...
 
-
 // Plugin linking
 
 #define STATIC_PLUGIN 1
 #define DYNAMIC_PLUGIN 2
 
-#define PLUGIN_ENABLED_STATIC(ID) \
-	(ENABLE_##ID && !PLUGIN_ENABLED_DYNAMIC(ID))
+#define PLUGIN_ENABLED_STATIC(ID) (ENABLE_##ID && !PLUGIN_ENABLED_DYNAMIC(ID))
```
Comment (sev-) [inline base/plugins.h]: And now it is harder to read...

File: base/plugins.h [hunk 9]
```diff
@@ -325,13 +312,16 @@ class PluginManager {
 public:
 	virtual ~PluginManager();
 
-	static void destroy() { delete _instance; _instance = 0; }
```
Comment (sev-) [inline base/plugins.h]: It is interesting like you split this oneliner while adding few right above.

File: base/version.h [hunk 10]
```diff
@@ -23,10 +23,10 @@
 #ifndef BASE_VERSION_H
 #define BASE_VERSION_H
 
-extern const char *gScummVMVersion;		// e.g. "0.4.1"
```
Comment (sev-) [inline base/version.h]: This all was tab-aligned...

**Comment by csnover** [comment]
@8BitPimp are you planning on continuing to work on this patch?

**Comment by 8BitPimp** [comment]
@csnover Yes, I can redo these changes to address the issues raised by @sev- :)
Thanks for taking a look at the patch.

**Comment by sev-** [comment]
@8BitPimp Any update?

**Comment by sev-** [comment]
Still no updates?

**Comment by bonki** [comment]
@8BitPimp Are you planning on updating this PR?

**Comment by bonki** [comment]
Since there hasn't been any update for months I'm closing this for now, but feel free to reopen if you want to work on this PR again.
